rinja_parser/src/target.rs

```diff
@@ -195,19 +227,29 @@ fn collect_targets<'a, T>(
     Ok((i, (singleton, targets)))
 }
 
-fn only_one_rest_pattern(targets: Vec<Target<'_>>) -> Result<Vec<Target<'_>>, ParseErr<'_>> {
-    let snd_wildcard = targets
-        .iter()
-        .filter_map(|t| match t {
-            Target::Rest(s) => Some(s),
-            _ => None,
-        })
-        .nth(1);
-    if let Some(snd_wildcard) = snd_wildcard {
-        return Err(nom::Err::Failure(ErrorContext::new(
-            "`..` can only be used once per tuple pattern",
-            snd_wildcard,
-        )));
+fn only_one_rest_pattern<'a>(
+    targets: Vec<Target<'a>>,
+    allow_named_rest: bool,
+    type_kind: &str,
+) -> Result<Vec<Target<'a>>, ParseErr<'a>> {
+    let mut found_rest = false;
+
+    for target in &targets {
+        if let Target::Rest(s) = target {
+            if !allow_named_rest && s.inner.is_some() {
+                return Err(nom::Err::Failure(ErrorContext::new(
+                    format!("`@ ..` is only allowed in slice patterns"),
```

Comment:
The `format!()` is not needed anymore.

Reply:
Even clippy was complaining about it. :3